uk.co.compucorp.civicrm.hrleaveandabsences/tests/phpunit/api/v3/LeaveRequestTest.php

```diff
@@ -1227,4 +1228,169 @@ public function testDeleteAlsoDeletesLeaveRequestAndItsBalanceChangesFor() {
 
     $this->fail("Expected to not find the LeaveRequest with {$leaveRequest->id}, but it was found");
   }
+
+  public function testGetShouldOnlyReturnTheLeaveRequestsOfStaffMembersManagedByTheContactOnTheManagedByParam() {
+    $manager1 = ContactFabricator::fabricate();
+    $manager2 = ContactFabricator::fabricate();
+
+    $staffMember1 = ContactFabricator::fabricate();
+    $staffMember2 = ContactFabricator::fabricate();
+    $staffMember3 = ContactFabricator::fabricate();
+    $staffMember4 = ContactFabricator::fabricate();
+
+    // We need the contracts because LeaveRequest.get only returns Leave Requests
+    // overlapping contracts
+    HRJobContractFabricator::fabricate(
+      ['contact_id' => $staffMember1['id']],
+      ['period_start_date' => '2016-01-01']
+    );
+
+    HRJobContractFabricator::fabricate(
+      ['contact_id' => $staffMember2['id']],
+      ['period_start_date' => '2016-01-01']
+    );
+
+    HRJobContractFabricator::fabricate(
+      ['contact_id' => $staffMember3['id']],
+      ['period_start_date' => '2016-01-01']
+    );
+
+    HRJobContractFabricator::fabricate(
+      ['contact_id' => $staffMember4['id']],
+      ['period_start_date' => '2016-01-01']
+    );
+
```

Comment:
I don't think that that would be a good idea. It's just a coincidence (and a bit a laziness from my part) that all the contracts start at the same date. Having a method which creates contracts starting at 2016-01-01 is not very useful as it is too specific for just a few tests. 

With that said, instead of creating the method you have suggested, I changed the contract dates. Now each one start on a different day and we have some more "randomness" in the tests.